### PR TITLE
Bump golangci-lint, Docker and Go CI versions

### PR DESCRIPTION
**Pull request by thrasher-**
# PR Description

Updates our CI Go version depends to 1.15x and bumps golangci-lint to the latest version, including linter fixes.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested

- [x] go test ./... -race
- [x] golangci-lint run

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally and on Travis with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by xtda — CHANGES_REQUESTED** [review]
I have been able to test everything so far besides coinbase pro as i don't have orders on there 

linters pass 
docker image builds (for daemon web has some other issues outside this PR)

One small optional change for bithumb but everything else is good 🎉

**Review by shazbert — APPROVED** [review]
tACK! - Tested Bithumb, could not test CoinbasePro.

**Review by gloriousCode — APPROVED** [review]
The linting changes are all good. Docker in the context of switching to golang 1.15 is all good. So this gets a tACK!

It has highlighted that I cannot build the web portion under `docker-compose up --build`. To ensure I didn't have a local environmental issue, I was able to build other projects in docker that have a web container using npm. However this also occurs in `master` branch and is therefore out of scope.

File: exchanges/bithumb/bithumb.go [hunk 1]
```diff
@@ -157,7 +157,7 @@ func (b *Bithumb) GetAccountInformation(currency string) (Account, error) {
 
 	val := url.Values{}
 	if currency != "" {
-		val.Set("currency", currency)
+		val.Set("order_currency", currency)
```
[the remark this answers is not in the document]
Comment (thrasher-) [reply exchanges/bithumb/bithumb.go]: Done!

**Comment by thrasher-** [comment]
> The linting changes are all good. Docker in the context of switching to golang 1.15 is all good. So this gets a tACK!
> 
> It has highlighted that I cannot build the web portion under `docker-compose up --build`. To ensure I didn't have a local environmental issue, I was able to build other projects in docker that have a web container using npm. However this also occurs in `master` branch and is therefore out of scope.

Pushed a fix up to potentially address this

**Review by xtda — APPROVED** [review]
```shell
Successfully built 509ca6158463
Successfully tagged gocryptotrader_web:latest
Recreating gocryptotrader_daemon_1 ... done
Recreating gocryptotrader_web_1    ... done
```

docker changes work 🎉 

same as before tested besides coinbase as i have no active order history on them to test 
approved!

**Review by gloriousCode — APPROVED** [review]
Thanks for making those changes! I can now happily build using docker 
# 🐳 🐋

**Pull request merged**